### PR TITLE
DEVPROD-5143: Add @evg-ui/lib package

### PR DESCRIPTION
**Pull request by sophstad**
DEVPROD-5143

### Description

- Add new `@evg-ui/lib` package which will include components, constants, utils, etc.
  - Move two array utils and one React component into new package
  - Add TypeScript `paths` config to allow importing via, e.g., `@evg-ui/lib/components/TextInputWithGlyph`
- Use `@evg-ui` package scope for ES Lint
- Move lint-staged into `packages/`

### Testing

Add new `lib` build variant to Evergreen for testing. For now, run unit tests, linting, and typechecking. Storybook and snapshots will be added at some point too (after DEVPROD-5139 is done)

### CODE REVIEW

File: apps/parsley/vite.config.ts [hunk 1]
```diff
@@ -47,7 +47,6 @@ export default defineConfig({
       },
       // Exclude storybook stories from fast refresh.
       exclude: /\.stories\.tsx?$/,
-      fastRefresh: true,
```
Comment (sophstad) [inline apps/parsley/vite.config.ts:50]: This is the default and I believe the option is no longer supported by Vite

File: apps/parsley/src/pages/LogDrop/ParseLogSelect/index.tsx [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import { useState } from "react";
 import styled from "@emotion/styled";
+import { size } from "@evg-ui/lib/constants/tokens";
```
[the remark this answers is not in the document]
Comment (sophstad) [reply apps/parsley/src/pages/LogDrop/ParseLogSelect/index.tsx:3]: I think this makes sense to handle in DEVPROD-5148, left a note there!

File: apps/spruce/tsconfig.json [hunk 3]
```diff
@@ -2,10 +2,9 @@
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
     "baseUrl": "src",
-    "types": [
-      "vitest/globals",
-      "vitest/jsdom"
-    ]
+    "paths": {
+      "@evg-ui/lib/*": ["../../../packages/lib/src/*"]
```
[the remark this answers is not in the document]
Comment (sophstad) [reply apps/spruce/tsconfig.json:6]: This just follows the structure we use everywhere else with application code. It does simplify things like linting/tsconfig where we can say "look in the `src` directory"

File: packages/lint-staged/index.js [hunk 4]
```diff
@@ -1,8 +1,4 @@
 module.exports = {
   "**/*.{js,ts,tsx}": ["yarn eslint:staged", "yarn prettier"],
-  "src/gql/**/*.{graphql,gql}": [
-    "yarn eslint:staged",
-    "yarn prettier --parser graphql",
-  ], // For GraphQL files, run eslint and prettier, and gql schema check
```
[the remark this answers is not in the document]
Comment (sophstad) [reply packages/lint-staged/index.js:6]: Good catch, I meant to update the line above (`--parser` is [implicit](https://prettier.io/docs/en/options.html#parser))

File: packages/lib/.eslintrc.cjs [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (sophstad) [reply packages/lib/.eslintrc.cjs:1]: I do not—do you also see it on Spruce and Parsley's ES Lint files?

[the remark this answers is not in the document]
Comment (minnakt) [reply packages/lib/.eslintrc.cjs:1]: I actually do see them if I have the UI folder open but not if I just have the Spruce & Parsley folders open. Maybe I'll just ask you on Thursday. 😅

**Review by minnakt — APPROVED** [review]

**Pull request merged**